bin/docker-image-tool.sh

```diff
@@ -113,16 +115,23 @@ function build {
     KYUUBI_ROOT="$CTX_DIR/base"
   fi
 
-  # cp spark for kyuubi as submit client
-  # if user set -s(spark-provider), use if
-  # else use builtin spark
+  # mkdir spark-binary to cache spark
+  # clean cache if spark-binary exists
   if [[ ! -d "$KYUUBI_ROOT/spark-binary" ]]; then
     mkdir "$KYUUBI_ROOT/spark-binary"
+  else
+    rm -rf "$KYUUBI_ROOT/spark-binary/*"
   fi
-  if [[ ! -d "$SPARK_HOME" ]]; then
-    error "Cannot found dir $SPARK_HOME, you must configure SPARK_HOME correct."
+  # if with spark image means, kyuubi won't copy spark from local.
+  # In this case, we just pass SPARK_HOME without copy.
+  if [[ "${WITHSPARKIMAGE}" != "false" ]]; then
+    BUILD_ARGS+=(--build-arg spark_home=$SPARK_HOME)
```

Comment:
can we also do `cp -r "$SPARK_HOME/" "$KYUUBI_ROOT/spark-binary/` here ? so we can make `SPARK_HOME` 
 and `SPARK_BINARY` consisent in `dockerfile`.

Reply:
Here, meams kyuubi use spark which from BASE_IMAGE. Spark already in Image, we can not move here.

Reply:
so this `spark_home` is the directory in docker rather than in real world ?

Reply:
We can think of it as having two `SPARK_HOME` on the outside and inside the container. SPARK_HOME inside the container is `/opt/spark` by default, which is the folder inside the container that you COPY to.